x/programs/runtime/import_program.go

```diff
@@ -49,18 +58,16 @@ func NewProgramModule(r *WasmRuntime) *ImportModule {
 
 				result, err := r.CallProgram(
 					context.Background(),
-					&newInfo)
-				if err != nil {
-					return nil, err
-				}
```

Comment:
Yes that's the reason, I'm not really sure how to rewrite it though since we need to be careful not to reassign the `err` variable.

Reply:
I think it just deserves a comment saying "need to handle fuel before dealing with the error" or whatever

Reply:
As I removed the `err` from the above `r.CallProgram`, I just replace the `fuelErr` by the usual `err`. I think that it doesn't need an err anymore since the flow is the same as previous

Reply:
Please see my other comment that starts with 
>The logic to handle the trap error should exist here

I can't link until I finish the review